comment/static/js/comment.js

```diff
@@ -1,320 +1,386 @@
+        let commentDOM = new DOMParser().parseFromString(data, "text/html");
+        return commentDOM.querySelector(selector);
+    };
+
+    let convertFormDataToURLQuery = formData => {
+        return Object.keys(formData).map((key) => {
+          return encodeURIComponent(key) + '=' + encodeURIComponent(formData[key]);
+        }).join('&');
+    };
+
+    // create new comment
+    let submitCommentCreateForm = form => {
+        let formButton = form.querySelector("button");
+        let url = form.getAttribute('data-url') || window.location.href;
+        let formData = serializeObject(form);
+        // this step is needed to append the form data to request.POST
+       let formDataQuery = convertFormDataToURLQuery(formData);
+
+       fetch(url, {
+           method: 'POST',
+            headers: {
+                'X-Requested-With': 'XMLHttpRequest',
+                'X-CSRFToken': csrfToken,
+                'Content-Type': 'application/x-www-form-urlencoded'
             },
+            body: formDataQuery
+       }).then(response => {
+            return response.text();
+        }).then(data => {
+           // parent comment
+            if (formButton.getAttribute('value') === 'parent') {
+                // reload all comments only when posting parent comment
+                document.getElementById("comments").outerHTML = data;
+            } else {
+                // child comment
+                let childComment = stringToDom(data, '.js-child-comment');
+                form.parentNode.insertBefore(childComment, form);
+                // update number of replies
+                let reply = form.parentElement.previousElementSibling.querySelector(".js-reply-link");
+                let replyNumberElement = form.parentElement.previousElementSibling.querySelector(".js-reply-number");
+                let replyNum = Number(replyNumberElement.innerText) + 1;
+                replyNumberElement.textContent = replyNum.toString();
+                if (replyNum > 1) {
+                    reply.textContent = "Replies";
+                } else {
+                    reply.textContent = "Reply";
+                }
+                commentCount(1);
+            }
+            formButton.setAttribute('disabled', 'disabled');
+            let elements = document.getElementsByClassName("js-comment-input");
+            Array.prototype.forEach.call(elements, element => {
+                element.setAttribute("style", "height: 31px;");
+                element.value = '';
+            });
+            // remove pagination query when posting a new comment
+            let uri = window.location.toString();
+            if (uri.indexOf("?") > 0) {
+                let clean_uri = uri.substring(0, uri.indexOf("?"));
+                window.history.replaceState({}, document.title, clean_uri);
+            }
+        }).catch(error => {
+            console.log(error);
+            alert("Unable to post your comment!, please try again");
```

Comment:
rather than using this, we may use the info function defined [here](https://github.com/Radi85/Comment/pull/49/files#diff-a13a7f5b55d01dc86d2535026a68b770R383)

Reply:
also, maybe somehow we can display the error phase from the backend here so that translated strings are shown here.